src/search_renderer.cpp

```diff
@@ -90,14 +95,20 @@ kainjow::mustache::data buildQueryData
 (
   const std::string& searchProtocolPrefix,
   const std::string& pattern,
+  const std::string& lang,
   const std::set<std::string>& bookIds
 ) {
   kainjow::mustache::data query;
   query.set("pattern", kiwix::encodeDiples(pattern));
   std::ostringstream ss;
   ss << searchProtocolPrefix << "?pattern=" << urlEncode(pattern);
-  for (auto& bookId: bookIds) {
-    ss << "&books.id="<<urlEncode(bookId);
+  if(!lang.empty()) {
+    query.set("lang", kiwix::encodeDiples(lang));
+    ss << "?books.query.lang=" << urlEncode(lang);
```

Comment:
`?` is already introduced by `?pattern=`. Change `?` → `&`

Reply:
Done